core/src/main/scala/akka/kafka/scaladsl/Consumer.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.kafka.scaladsl
+
+import scala.concurrent.Future
+import akka.Done
+import akka.stream.scaladsl.Source
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import akka.kafka.internal.ConsumerStage
+import akka.NotUsed
+import akka.dispatch.ExecutionContexts
+import scala.concurrent.ExecutionContext
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.clients.consumer.OffsetAndMetadata
+import akka.kafka.internal.ConsumerStage.CommittableOffsetBatchImpl
+
+object Consumer {
+
+  trait Message[K, V] {
```

Comment:
this makes it possible to implement `plainSouce` and `committableSouce` with the same `GraphStage` and not do additional copy of objects. We could use `ConsumerRecord` for the `plainSource` but then I think we should have two variants of the `GraphStage`. That should be possible.